src/sqlfluff/rules/L057.py

```diff
@@ -62,7 +64,11 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
             # Evaluate unquoted identifiers.
             if identifiers_policy_applicable(
                 self.unquoted_identifiers_policy, context.parent_stack
-            ) and not (context.segment.raw.replace("_", "").isalnum()):
+            ) and not (
+                context.segment.raw.replace("_", "")
```

Comment:
Nitpick: IIUC, we are ignoring occurrences of `_` as well as `additional_allowed_identifiers`, but doing so in two different ways. Think it might be clearer to incorporate the `_` into the other code somehow? One thought: set the default value of `additional_allowed_identifiers` to `_`. 

Also, it looks like this same (increasingly complex) code is repeated 3 times in the rule. Think we could avoid this duplication?

Reply:
I think I'd prefer to keep the underscores in there. To me they should always be allowed. Petty Python doesn't have an `isalnum` equivalent that include them.

Also means we can skip the slightly more complex additional logic if it's not set (which it isn't by default) which would be a slight performance gain... maybe.

However I have refactored the code to avoid duplicating it. Let me know what you think.